[kbn-evals] Doc trigger on-demand evals (#272814)
elastic/kibana: x-pack/platform/packages/shared/kbn-evals/README.md

---

### 1.3 On-demand evals

> Run a suite on any branch without a PR via the CI Stats trigger at [https://ci-stats.kibana.dev/trigger_evals_runner](https://ci-stats.kibana.dev/trigger_evals_runner).

Reviewer: Since this allows PRs too, I feel like this title doesn't match anymore. 
How about?

```suggestion
Run a suite on a branch or on an open/draft PR via the CI Stats trigger at [https://ci-stats.kibana.dev/trigger_evals_runner](https://ci-stats.kibana.dev/trigger_evals_runner).
```

> 1. Choose a branch or open PR

Reviewer: What we want to convey here is "Pick a PR" right, not open PR
```suggestion
1. Choose a branch or pick an open/draft PR
```

> 2. Select the eval suite and model groups

Reviewer: ```suggestion
2. Select the eval suite and one or multiple models
```

> 3. Optionally set a test name filter (`EVAL_GREP`), repetitions (`EVALUATION_REPETITIONS`), judge connector, or Kibana build id to reuse

Reviewer: ```suggestion
3. Optionally set a test name filter (`EVAL_GREP`), repetitions (`EVALUATION_REPETITIONS`), judge model, and Kibana build id to reuse
```

4. Review the build parameters and trigger

> Note: you can trigger On-demand evals directly on Buildkite, open [kibana-evals-on-demand](https://buildkite.com/elastic/kibana-evals-on-demand), click **New build**, Add environment variables:

Reviewer: ```suggestion
Note: you can trigger on-demand evals directly on Buildkite, open [kibana-evals-on-demand](https://buildkite.com/elastic/kibana-evals-on-demand), click **New build**, Add environment variables:
```
